### PR TITLE
Merge mcbs branch into master

### PR DESCRIPTION
**Pull request by cheesesashimi**
This aims to merge the `mcbs` branch, to incorporate the majority of our current CoreOS layering work into master. As of this writing, the current FCOS / RHCOS images do not have the OCI-aware version of `rpm-ostree` baked into them. This means the CI runs attached to this PR are expected to fail until they're available, hence why this is a draft for now.

Once the FCOS / RHCOS images are tagged into 4.11 and any CI failures introduced by this PR are addressed, it can be merged after passing an enhanced test suite supplied by the TRT team.

### CODE REVIEW

File: pkg/daemon/rpm-ostree_test.go [hunk 1]
```diff
@@ -46,3 +54,39 @@ func (r RpmOstreeClientMock) GetStatus() (string, error) {
 func (r RpmOstreeClientMock) GetBootedDeployment() (*RpmOstreeDeployment, error) {
 	return &RpmOstreeDeployment{}, nil
 }
+
+func TestParseVersion(t *testing.T) {
```
Comment (kikisdeliveryservice) [inline pkg/daemon/rpm-ostree_test.go:58]: what is this testing exactly?

File: pkg/daemon/rpm-ostree.go [hunk 2]
```diff
@@ -100,6 +115,60 @@ func (r *RpmOstreeClient) loadStatus() (*rpmOstreeState, error) {
...
+		n, err := strconv.Atoi(s)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to parse %s: %v", s, err)
+		}
+		r = append(r, n)
+	}
+	return r, nil
+}
+
+func validateVersion(current rpmOstreeVersionOuter) error {
```
Comment (kikisdeliveryservice) [inline pkg/daemon/rpm-ostree.go:130]:
```suggestion
func validateRpmOstreeVersion(current rpmOstreeVersionOuter) error {
```

clearer and in line with below function

File: pkg/daemon/rpm-ostree.go [hunk 3]
```diff
@@ -100,6 +115,60 @@ func (r *RpmOstreeClient) loadStatus() (*rpmOstreeState, error) {
 	return &rosState, nil
 }
 
+func parseVer(s string) ([]int, error) {
```
Comment (kikisdeliveryservice) [inline pkg/daemon/rpm-ostree.go:118]:
```suggestion
func parseRpmOstreeVer(s string) ([]int, error) {
```

can you also add a docstring to this function bc it's really not clear what it is doing.

File: test/helpers/utils.go [hunk 4]
```diff
@@ -166,6 +166,22 @@ func WaitForRenderedConfigs(t *testing.T, cs *framework.ClientSet, pool string,
 	return renderedConfig, nil
 }
 
+func WaitForBuild(t *testing.T, cs *framework.ClientSet, build string) {
```
Comment (kikisdeliveryservice) [inline test/helpers/utils.go:169]: add docstring

File: Makefile [hunk 5]
```diff
@@ -98,7 +98,7 @@ Dockerfile.rhel7: Dockerfile Makefile
 
 # This was copied from https://github.com/openshift/cluster-image-registry-operator
 test-e2e:
-	go test -tags=$(GOTAGS) -failfast -timeout 90m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e/
+	go test -tags=$(GOTAGS) -failfast -timeout 100m -v$${WHAT:+ -run="$$WHAT"} ./test/e2e/
```
Comment (kikisdeliveryservice) [inline Makefile:101]: can we add these tests as a new e2e like we did for single node below? we'll want to test more and there isn't a reasonable way to add them to this tests suite as it's already 90m. We previously spent a lot of time getting these tests down to around 90 and we shouldn't let them go higher. It's easier to do this now as opposed to later.

File: go.mod [hunk 6]
```diff
@@ -45,6 +45,7 @@ require (
 	k8s.io/component-base v0.23.0
 	k8s.io/kubectl v0.23.0
 	k8s.io/kubelet v0.23.0
+	k8s.io/kubernetes v1.13.0
```
Comment (kikisdeliveryservice) [inline go.mod:48]: is v1.13.0 correct?? that's 10 versions behind the last release (1.23)

**Review by kikisdeliveryservice — COMMENTED** [review]
made a first pass with some requests/questions.

File: Makefile (hunk 5 above)
Comment (cheesesashimi) [reply Makefile:101]: I wholeheartedly agree with this statement. If we do that, we'll also need a PR to https://github.com/openshift/release/blob/master/ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml to add another test step target.

Comment (mkenigs) [reply Makefile:101]: I'm not even really sure the test that required that bump should get merged to master. It's a PoC, not actually testing MCO code

**Comment by cgwalters** [comment]
OK, the [current 4.11 nightly](https://openshift-release.apps.ci.l2s4.p1.openshiftapps.com/releasestream/4.11.0-0.nightly/release/4.11.0-0.nightly-2022-02-27-122819) should have what we need!

File: Makefile (hunk 5 above)
Comment (cheesesashimi) [reply Makefile:101]: That's a good point. For now, we can probably remove that commit from this PR. I need to find a new location for it though.

**Pull request closed without merging**